test/jdk/javax/swing/JFileChooser/CustomApproveButtonTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import java.awt.Component;
+import java.awt.Container;
+
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+/*
+ * @test
+ * @bug 8299522
+ * @key headful
+ * @summary JFileChooser Test to check for FileChooser's Approve Button Text
+ *          when CUSTOM_DIALOG type is set.
+ * @run main CustomApproveButtonTest
+ */
+
+public class CustomApproveButtonTest {
+    private JFrame frame;
+
+    public static void main(String[] args) throws Exception {
+        String aquaLAF = "com.apple.laf.AquaLookAndFeel";
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                UIManager.LookAndFeelInfo[] lookAndFeel = UIManager.getInstalledLookAndFeels();
+                for (UIManager.LookAndFeelInfo look : lookAndFeel) {
+                    if (look.getClassName().equals(aquaLAF)) {
+                        continue;
+                    }
+                    new CustomApproveButtonTest(look.getClassName());
+                }
+            }
+        });
+        System.out.println("Test Pass!");
+    }
+
+    private CustomApproveButtonTest(String lookAndFeel) {
+        System.out.println("Testing Look & Feel : " + lookAndFeel);
+        frame = new JFrame("CustomApproveButtonTest");
+
+        try {
+            UIManager.setLookAndFeel(lookAndFeel);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to set " + lookAndFeel +
+                    " LookAndFeel. Error : " + e);
+        }
```

Comment:
You should probably quietly ignore the Look-and-Feel if `UnsupportedLookAndFeelException` is thrown.

Reply:
Updated.